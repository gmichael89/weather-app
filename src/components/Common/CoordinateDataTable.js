import React from 'react'

export default function CoordinateDataTable(coords) {

    const { lat, lng } = coords;

    if (!lat) {
        return ''
    }

    return (
        <section>
            <h2>Your current position</h2>
            <table>
                <tbody>
                    <tr>
                        <td>Latitude:</td>
                        <td>{lat}</td>
                    </tr>
                    <tr>
                        <td>Longitude:</td>
                        <td>{lng}</td>
                    </tr>
                </tbody>
            </table>
        </section>
    );
}
