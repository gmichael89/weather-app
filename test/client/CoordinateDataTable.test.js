import React from 'react';
import CoordinateDataTable from '../../src/components/Common/CoordinateDataTable';
import renderer from 'react-test-renderer';

test('Table doesn\'t render when no input is provided.', () => {

    const component = renderer.create(
        <CoordinateDataTable />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});

test('Table renders with provided latitude and longitude.', () => {

    const coords = {
        lat: 53.4186504,
        lng: -2.2391706
    };

    const component = renderer.create(
        CoordinateDataTable(coords)
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

});
