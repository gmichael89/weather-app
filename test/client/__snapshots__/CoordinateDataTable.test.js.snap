// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table doesn't render when no input is provided. 1`] = `""`;

exports[`Table renders with provided latitude and longitude. 1`] = `
<section>
  <h2>
    Your current position
  </h2>
  <table>
    <tbody>
      <tr>
        <td>
          Latitude:
        </td>
        <td>
          53.4186504
        </td>
      </tr>
      <tr>
        <td>
          Longitude:
        </td>
        <td>
          -2.2391706
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;
